### ISSUE TITLE
The results on the sidd raw-RGB validation set cannot be reproduced as high as 51.06dB

### ISSUE THREAD

**Issue by zejinwang** [TaoHuang2018/Neighbor2Neighbor#8]
Thank you for your recent open source.
I reproduced the experiment on SIDD raw-RGB, but it can only achieve about 49.07dB. I know there are problems with some details and hope to get your help.
Here is my code：
`
        for valid_name, valid_data in valid_dict.items():
            psnr_result = []
            ssim_result = []
            num_img, num_block, valid_noisy, valid_gt = valid_data
            for idx in range(num_img):
                for idy in range(num_block):
                    im = valid_gt[idx, idy][:, :, np.newaxis]
                    noisy_im = valid_noisy[idx, idy][:, :, np.newaxis]

                    origin255 = im.copy() * 255.0
                    origin255 = origin255.astype(np.uint8)
                    noisy255 = noisy_im.copy() * 255.0
                    noisy255 = noisy255.astype(np.uint8)
                    # padding to square
                    H = noisy_im.shape[0]
                    W = noisy_im.shape[1]
                    val_size = (max(H, W) + 31) // 32 * 32
                    noisy_im = np.pad(
                        noisy_im,
                        [[0, val_size - H], [0, val_size - W], [0, 0]],
                        'reflect')
                    transformer = transforms.Compose([transforms.ToTensor()])
                    noisy_im = transformer(noisy_im)
                    noisy_im = torch.unsqueeze(noisy_im, 0)
                    noisy_im = noisy_im.cuda()
                    # pack raw data
                    noisy_im = space_to_depth(noisy_im, block_size=2)
                    with torch.no_grad():
                        prediction = network(noisy_im)
                        # unpack raw data
                        prediction = depth_to_space(prediction, block_size=2)
                        prediction = prediction[:, :, :H, :W]
                    prediction = prediction.permute(0, 2, 3, 1)
                    prediction = prediction.cpu().data.clamp(0, 1).numpy()
                    prediction = prediction.squeeze(0)
                    pred255 = np.clip(prediction * 255.0 + 0.5, 0,
                                        255).astype(np.uint8)
                    # calculate psnr
                    cur_psnr = calculate_psnr(origin255.astype(np.float32),
                                                pred255.astype(np.float32))
                    psnr_result.append(cur_psnr)
                    cur_ssim = calculate_ssim(origin255.astype(np.float32),
                                                pred255.astype(np.float32))
                    ssim_result.append(cur_ssim)

                    # visualization
                    save_path = os.path.join(
                        validation_path,
                        "{}_{:03d}-{:03d}-{:03d}_clean.png".format(
                            valid_name, idx, idy, epoch))
                    Image.fromarray(origin255.squeeze()).save(save_path)
                    save_path = os.path.join(
                        validation_path,
                        "{}_{:03d}-{:03d}-{:03d}_noisy.png".format(
                            valid_name, idx, idy, epoch))
                    Image.fromarray(noisy255.squeeze()).save(save_path)

                    save_path = os.path.join(
                        validation_path,
                        "{}_{:03d}-{:03d}-{:03d}_denoised.png".format(
                            valid_name, idx, idy, epoch))
                    Image.fromarray(pred255.squeeze()).save(save_path)

            psnr_result = np.array(psnr_result)
            avg_psnr = np.mean(psnr_result)
            avg_ssim = np.mean(ssim_result)
            log_path = os.path.join(validation_path,
                                    "A_log_{}.csv".format(valid_name))
            with open(log_path, "a") as f:
                f.writelines("{},{},{}\n".format(epoch, avg_psnr, avg_ssim))
`

**Comment by ghost** [comment]
请问你是如何测试这篇文章的代码的

**Comment by zejinwang** [comment]
> 请问你是如何测试这篇文章的代码的
sRGB部分在uint8范围测试
rawRGB在[0,1]范围测试
测试后没有问题

**Comment by leonmakise** [comment]
请问你是自己在sidd上面训练了一遍吗，作者似乎没有提供SIDD上的预训练模型？
